[Aryan Sarswat] iP

AryanSarswat opened the pull request:
### DukeTransformed
> Beep Boop Beep Boop  :robot:

DukeTransformed is an **upgraded** version of Duke which consist of many _new_ features such as:

- [x] Managing Tasks
- [x] Managing Deadlines
- [x] Managing Events
- [x] Finding Tasks
- [x] Mark Tasks as Done and Not Done
- [x] Deleting Tasks
- [x] GUI

Duke transformed is:

1. Free
2. easy to learn
3. ~~FAST~~ EXTREMELY Fast to use

Download it using this [link](https://github.com/AryanSarswat/ip/releases/tag/A-Jar)

If you have experience in java you can do the following:
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

AryanSarswat committed:
Implemented Level 1

AryanSarswat committed:
Implemented Level 2

AryanSarswat committed:
Implemented Level-3

AryanSarswat committed:
Implemented Level 4

AryanSarswat committed:
Implemented Automated Testing and Level 5

AryanSarswat committed:
Implemented Level-6

AryanSarswat committed:
Encapsulated logic and refactored the indentation

AryanSarswat committed:
Minor typo fixes

AryanSarswat committed:
Finished Level-7

AryanSarswat committed:
Finished Level-8

AryanSarswat committed:
Modified Tests

AryanSarswat committed:
Merge branch 'branch-Level-7'

AryanSarswat committed:
Merge branch 'branch-Level-8'

AryanSarswat committed:
Refactors and organised everything with more OOP

AryanSarswat committed:
Organized into packages

AryanSarswat committed:
Added Junit tests for Task and Parser

AryanSarswat committed:
Made Jar file

AryanSarswat committed:
Added JavaDocs for all Public functions

AryanSarswat committed:
Changed to code to be in accordance with Java coding standard

AryanSarswat committed:
Added find functionality

AryanSarswat committed:
Merge tag 'A-CodingStandard'

Changed code to conform to java coding standards

AryanSarswat committed:
Bug fixes

AryanSarswat committed:
Merge tag 'Level-9'

Finished adding find functionality

AryanSarswat committed:
Corrected some formatting issues

AryanSarswat committed:
Bug Fix

AryanSarswat committed:
Made project into a gradle project

AryanSarswat committed:
Merge tag 'A-Gradle'

Added Gradle support

AryanSarswat committed:
Add a GUI to Duke

AryanSarswat committed:
Size update for GUI

AryanSarswat committed:
Add checkstyle.xml file

AryanSarswat committed:
Fixed bug with text being truncated in GUI

AryanSarswat committed:
Re-arrange duke files in to sub packages

hanhuiice commented on src/main/java/duke/Commands/Command.java:
```diff
@@ -0,0 +1,174 @@
+    public static String execute(String[] processedInput, TaskList listOfTasks) {
+        String toPrint = "";
+        try {
+            switch (processedInput[0]) {
```
The syntax for the switch statement follows the given guideline. 👍

hanhuiice commented on src/main/java/duke/Commands/Parser.java:
```diff
@@ -0,0 +1,110 @@
+ */
+public class Parser {
+
+    private static boolean isInteger(String str) {
```
The boolean method is named properly, I can understand what it is suppose to do easily. 👍

hanhuiice commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+     * Function to enable the bot and start interaction with the user.
+     */
+    public void run() {
+        this.ui.greet();
```
Maybe `this` keyword is not necessary here.

hanhuiice commented on src/main/java/duke/Tasks/Task.java:
```diff
@@ -0,0 +1,101 @@
+package duke.Tasks;
+
+import java.time.LocalDateTime;
```
Imports are listed nicely, easier to see what the class might be doing.

hanhuiice reviewed (commented):
Nice your syntax looking clean.
Just here for a visit.

FYimu commented on src/main/java/duke/Commands/Command.java:
```diff
@@ -0,0 +1,174 @@
+import duke.Tasks.Event;
+import duke.Tasks.Task;
+import duke.Tasks.TaskList;
+import duke.Tasks.ToDos;
```
I personally feel like it is uncommon to name a class using plural forms, because it is a bit strange if we say "create a ToDos instance"?

FYimu commented on src/main/java/duke/Commands/Command.java:
```diff
@@ -0,0 +1,174 @@
+        }
+        return toPrint;
+    }
+
```
You might want to consider writing some classes inherited from Command classes to abstract different methods, e.g. MarkCommand etc? It might help to enhance code readability 😉

FYimu commented on src/main/java/duke/Storage/Storage.java:
```diff
@@ -0,0 +1,130 @@
+        } catch (DukeException ex) {
+            System.out.println("IOException while trying to read from file");
+        } catch (IOException ex) {
+            System.out.println("Error in storage format");
```
Since UI (not sure if GUI serves the same purpose as I have not touched level 10) is responsible for interactions with users, would you like to consider leaving all the `System.out.println()` to ui? e.g. replace here with `ui.showError(msg)` so that all messages displayed are directly from ui 😜

FYimu commented on src/main/java/duke/Storage/Storage.java:
```diff
@@ -0,0 +1,130 @@
+        try {
+            BufferedWriter writer = new BufferedWriter(new FileWriter(this.filepath
+                                                        + File.separator + this.filename));
+            for (Task i : listOfTasks.getList()) {
```
You might want to consider using more descriptive variable name here? e.g. `Task task : listOfTasks.getList()`, because we normally use `for (int i = 0; i < num; i++)` because `i` refers to index in the iteration, but here it is a Task instance, so it might be better to use a more descriptive name :>

FYimu commented on src/main/java/duke/Tasks/Event.java:
```diff
@@ -0,0 +1,61 @@
+    /**
+     * Constructor for Event task.
+     *
+     * @param task description of the event
```
JavaDoc description needs to start with a capital letter~
e.g. 
```
* @param task Description of the event
```

FYimu reviewed (commented):
Hii! I like how you wrote JavaDoc for all public methods and classes, making your code clear and highly readable. It was a pleasant experience to read your code 😉

AryanSarswat committed:
Code Readability enhancements

AryanSarswat committed:
Add assertions to key points in the code

AryanSarswat committed:
Refactor code to fit Code Quality guidelines

AryanSarswat committed:
Merge pull request #2 from AryanSarswat/branch-A-Assertions

Add assertions to key points in the code

AryanSarswat committed:
Merge branch 'master' into branch-A-CodeQuality

AryanSarswat committed:
Merge pull request #1 from AryanSarswat/branch-A-CodeQuality

Refactor code to fit Code Quality guidelines

AryanSarswat committed:
Add help command

AryanSarswat committed:
Storage prints out statements instead of the UI

The Storage class itself contained System.out.println() messages
whearas this is the task of the UI to print out these messages

Refactoring the code such that the UI handles all the printing
task will be cleaner and will improve Code Quality

AryanSarswat committed:
Some parts of the Code did not follow module's Code Guidelines

It is important to standardise coding style as it imporves
readbility of code while doing peer reviews and also helps
preventing compatability issues.

Some of the lines in Commands.java were over 120 lines, the
recommended limit for this module is 120 lines thus, the particular
line in question was intended.

AryanSarswat committed:
Incoporate iP code quality feedback from automated anaylsis

There were several inconsistencies with code quality which were fixed
to follow the module's coding standard.

Following the coding standard will lead to better code readability,
and possible higher marks for my iP in this module.

AryanSarswat committed:
Font change and made images rounded

AryanSarswat committed:
Set theme jekyll-theme-tactile

AryanSarswat committed:
Set theme jekyll-theme-cayman

AryanSarswat committed:
find command bug fix

AryanSarswat committed:
Set theme jekyll-theme-tactile

AryanSarswat committed:
remove redundant file

AryanSarswat committed:
Pull website template code

Merge branch 'master' of https://github.com/AryanSarswat/ip

AryanSarswat committed:
Implement User Guide

AryanSarswat committed:
Set theme jekyll-theme-hacker

AryanSarswat committed:
Set theme jekyll-theme-dinky

AryanSarswat committed:
Set theme jekyll-theme-modernist

AryanSarswat committed:
Set theme jekyll-theme-merlot

AryanSarswat committed:
Update to UserGuide

AryanSarswat committed:
Pull github pages update

Merge branch 'master' of https://github.com/AryanSarswat/ip

AryanSarswat committed:
Code quality improvments
